### PR TITLE
[2020-07-30 10:52:42] Rest Api task 1

### PR DESCRIPTION
**Pull request by vishakraman**

### CODE REVIEW

File: web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   *   Throws exception expected.
+   */
+ public function get() {
+    $json_array = array(
+      'data' => array()
+    );
+    $nids = \Drupal::entityQuery('node')->condition('type','api')->execute();
+    $nodes =  \Drupal\node\Entity\Node::loadMultiple($nids);
+    foreach ($nodes as $node) {
+      dsm($node);
```
Comment (girish06) [inline web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php:89]: remove dsm

File: web/modules/custom/form/config/install/rest.resource.demo_resource.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+id: example_get_rest_resource
```
Comment (girish06) [inline web/modules/custom/form/config/install/rest.resource.demo_resource.yml:1]: can you please explain why this yml is required

**Review by girish06 — CHANGES_REQUESTED** [review]

File: web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      $json_array['data'][] = array(
+        'type' => $node->get('type')->target_id,
+        'id' => $node->get('nid')->value,
+        'attributes' => array(
+          'title' =>  $node->get('title')->value,
+          'body' => $node->get('body')->value,
+          'date' => $node->get('field_date')->value,
+          'boolean' => $node->get('field_bool')->value,
+          'description' => $node->get('field_description')->value,
+          'image'=>$node->field_image_api->entity->getFileUri(),
```
Comment (girish06) [inline web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php:99]: do validate

File: web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   */
+ public function get() {
+    $json_array = array(
+      'data' => array()
+    );
+    $nids = \Drupal::entityQuery('node')->condition('type','api')->execute();
```
Comment (girish06) [inline web/modules/custom/form/src/Plugin/rest/resource/ExampleGetRestResource.php:86]: only published content

**Review by girish06 — CHANGES_REQUESTED** [review]

**Comment by vishakraman** [comment]
Fixed and pushed

**Review by girish06 — APPROVED** [review]

**Pull request merged**